portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase

```diff
@@ -553,6 +555,123 @@ definition {
+			var universityId = ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "universities",
+				name = "Liferay University");
+		}
+
+		task ("Given subject entry created") {
+			var subjectId = ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "subjects",
+				name = "Liferay Foundations");
+		}
+
+		task ("Given student entry created") {
+			var studentId = ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "students",
+				name = "Liferay Employee");
+		}
+
+		task ("Given I relate subject entry with university entry through subjects PUT endpoint") {
+			ObjectDefinitionAPI.relateObjectEntries(
+				en_US_plural_label = "subjects",
+				objectEntry1 = "${subjectId}",
+				objectEntry2 = "${universityId}",
+				relationshipName = "universitiesSubjects");
+		}
+
+		task ("Given I relate student entry with university entry through students PUT endpoint") {
+			ObjectDefinitionAPI.relateObjectEntries(
+				en_US_plural_label = "students",
+				objectEntry1 = "${studentId}",
+				objectEntry2 = "${universityId}",
+				relationshipName = "universitiesStudents");
+		}
+
+		task ("When I call universities GET endpoint with nestedFields=universitiesSubjects") {
+			var getObjectsWithNestedFieldReponse1 = ObjectDefinitionAPI.getObjectsWithNestedField(
+				nestedField = "universitiesSubjects",
+				objects = "universities");
+		}
+
+		task ("Then I receive correct information about the universitiesSubjects") {
+			ObjectDefinitionAPI.assertNestedFieldDetail(
+				expectedValue = "${subjectId}",
+				getObjectsWithNestedFieldReponse = "${getObjectsWithNestedFieldReponse1}",
+				nestedField = "universitiesSubjects",
+				objectEntryId = "${universityId}",
+				objectField = "id");
+
+			ObjectDefinitionAPI.assertNestedFieldDetail(
+				expectedValue = "Liferay Foundations",
+				getObjectsWithNestedFieldReponse = "${getObjectsWithNestedFieldReponse1}",
+				nestedField = "universitiesSubjects",
+				objectEntryId = "${universityId}",
+				objectField = "name");
+		}
+
+		task ("When I call universities GET endpoint with nestedFields=universitiesStudents") {
+			var getObjectsWithNestedFieldReponse2 = ObjectDefinitionAPI.getObjectsWithNestedField(
+				nestedField = "universitiesStudents",
+				objects = "universities");
+		}
+
+		task ("Then I receive correct information about the universitiesStudents") {
+			ObjectDefinitionAPI.assertNestedFieldDetail(
+				expectedValue = "${studentId}",
+				getObjectsWithNestedFieldReponse = "${getObjectsWithNestedFieldReponse2}",
+				nestedField = "universitiesStudents",
+				objectEntryId = "${universityId}",
+				objectField = "id");
+
+			ObjectDefinitionAPI.assertNestedFieldDetail(
+				expectedValue = "Liferay Employee",
+				getObjectsWithNestedFieldReponse = "${getObjectsWithNestedFieldReponse2}",
+				nestedField = "universitiesStudents",
+				objectEntryId = "${universityId}",
+				objectField = "name");
+		}
+	}
+
```

Comment:
I don't see some test steps declared in the test. Are you sure you're automating the correct one?


```
Test Plan
Given university created
And Given subject created
And Given student
And Given manyToMany relationship universitySubjects created
And Given manyToMany relationship universityStudents created
And Given university entry created
And Given subject entry created
And Given student entry created
And Given in c/subjects I use PUT /{subjectId}/universitiesSubjects/{universityId}
And Given in c/students I use PUT /{studentId}/universitiesStudents/{universityId}
When I call /o/c/universities?nestedFields=universitiesSubjects,universitiesStudents
Then I receive correct information about the related objects
```